main-actions/src/main/scala/sbt/Tests.scala

```diff
@@ -531,8 +535,9 @@ object Tests {
       frameworks: Seq[Framework],
       analysis: CompileAnalysis,
       log: Logger
-  ): (Seq[TestDefinition], Set[String]) =
-    discover(frameworks flatMap TestFramework.getFingerprints, allDefs(analysis), log)
+  ): (TestDefinitionsWithFramework, Set[String]) = {
```

Comment:
Since `Tests` is under `sbt` package, I don't think we can break the binary compatibility during sbt 1.x series.